### PR TITLE
Readjust headers of document files

### PR DESCRIPTION
**Pull request by szarnyasg**
**This PR adjusts the headers in Markdown files so that they only use levels h2, h3, and h4.**

I hacked together a [script](https://gist.github.com/szarnyasg/9cc09f386abe171beee20a06aba0730b) that shows statistics of the pages. According to this, there are 33 pages with 3+ header levels.

```console
┌──────────────────────────────┬───────┬────────────┬─────────┬─────────────────┐
│             file             │ h_min │ max_header │ h_range │ h_unique_values │
│           varchar            │ int64 │   int64    │  int64  │      int32      │
├──────────────────────────────┼───────┼────────────┼─────────┼─────────────────┤
│ api/python/types.md          │     1 │          5 │       5 │               4 │
│ api/python/data_ingestion.md │     1 │          5 │       5 │               3 │
│ extensions/httpfs.md         │     1 │          4 │       4 │               4 │
│ api/python/overview.md       │     1 │          4 │       4 │               3 │
│ api/python/function.md       │     1 │          4 │       4 │               3 │
│ data/json/overview.md        │     1 │          4 │       4 │               4 │
│ sql/statements/copy.md       │     1 │          4 │       4 │               3 │
│ guides/index.md              │     2 │          5 │       4 │               3 │
│ sql/indexes.md               │     2 │          5 │       4 │               3 │
│ sql/query_syntax/with.md     │     2 │          5 │       4 │               4 │
│ guides/python/jupyter.md     │     1 │          3 │       3 │               3 │
│ api/r.md                     │     1 │          3 │       3 │               3 │
│ api/odbc/linux.md            │     1 │          3 │       3 │               3 │
│ api/odbc/macos.md            │     1 │          3 │       3 │               3 │
│ data/csv/overview.md         │     1 │          3 │       3 │               3 │
│ api/nodejs/reference.md      │     2 │          4 │       3 │               3 │
│ api/java.md                  │     2 │          4 │       3 │               3 │
│ api/cpp.md                   │     2 │          4 │       3 │               3 │
│ api/c/table_functions.md     │     2 │          4 │       3 │               3 │
│ api/c/appender.md            │     2 │          4 │       3 │               3 │
│ api/c/config.md              │     2 │          4 │       3 │               3 │
│ api/c/value.md               │     2 │          4 │       3 │               3 │
│ api/c/replacement_scans.md   │     2 │          4 │       3 │               3 │
│ api/c/connect.md             │     2 │          4 │       3 │               3 │
│ api/c/api.md                 │     2 │          4 │       3 │               3 │
│ api/c/query.md               │     2 │          4 │       3 │               3 │
│ api/c/data_chunk.md          │     2 │          4 │       3 │               3 │
│ api/c/prepared.md            │     2 │          4 │       3 │               3 │
│ api/c/types.md               │     2 │          4 │       3 │               3 │
│ sql/data_types/struct.md     │     2 │          4 │       3 │               3 │
│ sql/statements/pivot.md      │     2 │          4 │       3 │               3 │
│ sql/statements/unpivot.md    │     2 │          4 │       3 │               3 │
│ sql/window_functions.md      │     2 │          4 │       3 │               3 │
├──────────────────────────────┴───────┴────────────┴─────────┴─────────────────┤
│ 33 rows                                                             5 columns │
└───────────────────────────────────────────────────────────────────────────────┘
```

There are 4 pages with 4+ header levels.

```console
┌──────────────────────────┬───────┬────────────┬─────────┬─────────────────┐
│           file           │ h_min │ max_header │ h_range │ h_unique_values │
│         varchar          │ int64 │   int64    │  int64  │      int32      │
├──────────────────────────┼───────┼────────────┼─────────┼─────────────────┤
│ api/python/types.md      │     1 │          5 │       5 │               4 │
│ extensions/httpfs.md     │     1 │          4 │       4 │               4 │
│ data/json/overview.md    │     1 │          4 │       4 │               4 │
│ sql/query_syntax/with.md │     2 │          5 │       4 │               4 │
└──────────────────────────┴───────┴────────────┴─────────┴─────────────────┘
```

### CODE REVIEW

**Comment by szarnyasg** [comment]
I did a quick test with the following Markdown file:

```md
# h1

## h2

Some text in the chapter.

### h3

Some text in the section.

#### h4

Some text in the subsection.

##### h5

Some text in the subsubsection.

###### h6

Some text in the paragraph.

####### h7

Some text in the subparagraph.
```

When run through...

```bash
pandoc my.md --to=pdf --top-level-division=part --template=eisvogel2.tex --data-dir=. --output=out.pdf --pdf-engine=xelatex --variable book=true --variable numbersections=true
```

it produces:

Screenshot 2023-08-13 at 15 35 18

So while pandoc allows level 7 headers. (Meanwhile, `h7` is not a valid HTML element.) It's not that great but it gives us 3 headers level for each page to use (where they will be `h2`, `h3`, and `h4` (i.e. `##`, `###`, and `####` in Markdown). 

The PDF will be structured (with examples) as

- `#` Part: Docs
  - `##` Chapter: SQL
    - `###` Section: Statements
      - `####` Subsection: CREATE TABLE
        - ...and here comes the content from `create_table.md`

It's important (for me) to keep in mind that ~99% of the users will read the HTML, so that has to look good, the PDF is less important.

**Comment by szarnyasg** [comment]
Okay, this looks good now! I generated a PDF and its ToC also looked good (there was no random "jitter" when e.g. a h2 header is followed by an h4 immediately followed by an h3).

@Mause please let me know if this is good to merge.

File: docs/data/csv/overview.md [hunk 1]
```diff
@@ -29,14 +29,14 @@ COPY (SELECT * FROM ontime) TO 'flights.csv' WITH (HEADER 1, DELIMITER '|');
 COPY (FROM ontime) TO 'flights.csv' WITH (HEADER 1, DELIMITER '|');
 ```
 
-### CSV Loading
+## The DuckDB CSV reader
```
Comment (Mause) [inline docs/data/csv/overview.md:32]:
```suggestion
## CSV reader
```

**Review by Mause — CHANGES_REQUESTED** [review]

File: docs/api/cpp.md [hunk 2]
```diff
@@ -55,6 +55,7 @@ std::unique_ptr<QueryResult> result = prepare->Execute(12);
 
 ### Streaming Queries
 
+TODO
```
[the remark this answers is not in the document]
Comment (szarnyasg) [reply docs/api/cpp.md:58]: Yes, this is an empty section. But the TODO won't be there for long, I want to deal with all visible TODOs in #987.

**Pull request merged**